pkg/controller/deployment/deployment_controller.go

```diff
@@ -438,41 +438,124 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 	}
 
 	if d.Spec.Paused {
-		// TODO: Implement scaling for paused deployments.
-		// Don't take any action for paused deployment.
-		// But keep the status up-to-date.
-		// Ignore paused deployments
-		glog.V(4).Infof("Updating status only for paused deployment %s/%s", d.Namespace, d.Name)
-		return dc.syncPausedDeploymentStatus(d)
+		return dc.sync(d)
 	}
+
 	if d.Spec.RollbackTo != nil {
 		revision := d.Spec.RollbackTo.Revision
 		if _, err = dc.rollback(d, &revision); err != nil {
 			return err
 		}
 	}
 
+	if dc.isScalingEvent(d) {
+		return dc.sync(d)
```

Comment:
What happens if a rollout and scale event are observed in the same sync?  Looking at the logs, it looks like the deployment gets requeued several times as part of the event.